### PR TITLE
add email preference feature

### PR DESCRIPTION
**Pull request by holotsvan**
[https://github.com/ita-social-projects/GreenCity/issues/7380](https://github.com/ita-social-projects/GreenCity/issues/7380)

- Add UserNotificationPreference entity + update User entity
- Add EmailPreference enum
- Add annotation to check if user has some kind of email preference
- Add this annotation to methods
- Add tests

### CODE REVIEW

**Review by Maryna-511750 — APPROVED** [review]

File: core/.attach_pid57734
Comment (IliaWithHat) [inline core/.attach_pid57734:1]: What is this file and other empty files doing there? Is really needed?

Comment (holotsvan) [reply core/.attach_pid57734:1]: deleted them.

File: dao/src/main/java/greencity/entity/UserNotificationPreference.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package greencity.entity;
+
+import greencity.enums.EmailPreference;
+import jakarta.persistence.*;
```
Comment (KizerovDmitriy) [inline dao/src/main/java/greencity/entity/UserNotificationPreference.java]: wildcard imports

File: service/src/test/java/greencity/aspects/EmailPreferenceAspectTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.*;
```
Comment (KizerovDmitriy) [inline service/src/test/java/greencity/aspects/EmailPreferenceAspectTest.java]: same

File: service/src/test/java/greencity/service/NotificationServiceImplTest.java [hunk 3]
```diff
@@ -20,10 +20,7 @@
 import greencity.entity.Category;
 import greencity.entity.Place;
 import greencity.enums.EmailNotification;
-import greencity.message.GeneralEmailMessage;
-import greencity.message.HabitAssignNotificationMessage;
-import greencity.message.SendReportEmailMessage;
-import greencity.message.UserTaggedInCommentMessage;
+import greencity.message.*;
```
Comment (KizerovDmitriy) [inline service/src/test/java/greencity/service/NotificationServiceImplTest.java]: same

**Review by KizerovDmitriy — CHANGES_REQUESTED** [review]

**Review by IliaWithHat — APPROVED** [review]

File: service/src/test/java/greencity/service/NotificationServiceImplTest.java (hunk 3 above)
Comment (holotsvan) [reply service/src/test/java/greencity/service/NotificationServiceImplTest.java]: remove wildcards

**Review by KizerovDmitriy — APPROVED** [review]

**Pull request merged**